pkg/api_builder/lib/src/client_library_generator.dart

```diff
@@ -0,0 +1,211 @@
+              ..named = true
+              ..type = code.refer('Client', 'package:http/http.dart'),
+          ))
+          ..initializers.add(
+            code
+                .refer('_client')
+                .assign(
+                  code
+                      .refer('Client', 'package:api_builder/_client_utils.dart')
+                      .call([
+                    code.refer('baseUrl'),
+                  ], {
+                    'client': code.refer('client'),
+                  }),
+                )
+                .code,
+          )))
+        ..methods.addAll(handlers.map(_buildClientMethod)),
+    );
+
+final _parser = RegExp(r'([^<]*)(?:<([^>|]+)(?:\|([^>]*))?>)?');
+
+code.Method _buildClientMethod(
+  Handler h,
+) {
+  // Find url parameters
+  final params = _parser
+      .allMatches(h.route)
+      .map((m) => m[2])
+      .where((p) => p != null)
+      .toList();
+  // Create a url pattern that embeds parameters above
+  final urlPattern = h.route.replaceAllMapped(
+    _parser,
+    (m) => m[2] != null ? '${m[1]}\$${m[2]}' : m[1],
+  );
+
+  // Find if we have payload, and what type
+  final hasPayload = !h.element.parameters.last.type.isDartCoreString;
+  final payloadType = h.element.parameters.last.type;
+
+  // Check the return value of the method.
+  var retType = h.element.returnType;
+  // Unpack Future<T> and FutureOr<T> wrapping of responseType
+  if (retType.isDartAsyncFuture || retType.isDartAsyncFutureOr) {
+    retType = (retType as ParameterizedType).typeArguments.first;
+  }
+  final returnsResponse = _responseType.isAssignableFromType(retType);
+
+  // If return type is a Response, we create a client that return List<int>
+  code.Reference returnTypeRef = _referToType(retType);
+  if (returnsResponse) {
+    returnTypeRef = code.refer('List<int>');
+  }
+
+  return code.Method(
+    (b) => b
+      ..name = h.element.name
+      ..returns = code.TypeReference((b) => b
+        ..symbol = 'Future'
+        ..types.add(returnTypeRef))
+      ..requiredParameters.addAll([
+        for (final param in params)
+          code.Parameter(
+            (b) => b
+              ..name = param
+              ..type = code.refer('String'),
+          ),
+        if (hasPayload)
+          code.Parameter(
+            (b) => b
+              ..name = 'payload'
+              ..type = _referToType(payloadType),
+          )
+      ])
+      ..modifier = code.MethodModifier.async
+      ..body = returnsResponse ? Code.scope((r) => '''
+        return await _client.requestBytes(
+          verb: '${h.verb.toLowerCase()}',
+          path: '$urlPattern',
+          ${hasPayload ? 'body: payload.toJson(),' : ''}
+        );
+      ''') : Code.scope((r) => '''
+        return ${r(returnTypeRef)}.fromJson(await _client.requestJson(
+          verb: '${h.verb.toLowerCase()}',
+          path: '$urlPattern',
+          ${hasPayload ? 'body: payload.toJson(),' : ''}
+        ));
+      '''),
+  );
+}
+/*
```

Comment:
Remove?

Reply:
YES :)

sorry, it got a little late last night, hehe